spark/src/main/scala/com/cloudera/livy/spark/client/ClientSession.scala

```diff
@@ -80,6 +99,22 @@ class ClientSession(val sessionId: Int, createRequest: CreateClientRequest)
     new JobStatus(id, status.state, status.result, status.error)
   }
 
+
+  private def copyResourceToHDFS(dataStream: InputStream, name: String): URI = {
+    val filePath = new Path(sessionHome, name)
+    val outFile = fs.create(filePath, true)
+    val buffer = new Array[Byte](512 * 1024)
+    var read = -1
+    try {
+      while ({read = dataStream.read(buffer); read != -1}) {
```

Comment:
Looking at the API it doesn't seem hard to do, but given this is using the `FileSystem` API and it doesn't really expose NIO interfaces, it's probably not going to be any more efficient...

I'm mostly a little worried about the large buffer size.